GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1436,16 +1435,13 @@ void OnRevisionReadCompleted()
                     }
 
                     _gridView.LoadingCompleted();
+
                     await this.SwitchToMainThreadAsync(cancellationToken);
-                    if (refresh)
-                    {
-                        _gridView.Refresh();
-                    }
 
                     SetPage(_gridView);
                     _isRefreshingRevisions = false;
                     RevisionsLoaded?.Invoke(this, new RevisionLoadEventArgs(this, UICommands, getUnfilteredRefs, getStashRevs, forceRefresh));
-                    HighlightRevisionsByAuthor(GetSelectedRevisions());
+                    Refresh();
```

Comment:
Is this really necessary (other than when refresh is forced due to artificial was inserted)?

Reply:
Yes, `Refresh` must be called at least once. Otherwise, the column sizes are not correctly set, etc.
But it can be called earlier: 770c9e87a5ec6b5c08b421820d14f3983fab108c.
The last grid refresh should be triggered by the revisions loading - but only in a follow-up PR.

Reply:
Please add this as a comment so we don't have to scratch our heads two months from now.